unified-runtime/source/adapters/offload/enqueue.cpp

```diff
@@ -88,3 +90,66 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueUSMMemcpy2D(
     size_t, uint32_t, const ur_event_handle_t *, ur_event_handle_t *) {
   return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
 }
+
+UR_APIEXPORT ur_result_t UR_APICALL urEnqueueMemBufferRead(
+    ur_queue_handle_t hQueue, ur_mem_handle_t hBuffer, bool blockingRead,
+    size_t offset, size_t size, void *pDst, uint32_t numEventsInWaitList,
+    const ur_event_handle_t *phEventWaitList, ur_event_handle_t *phEvent) {
+
+  // Ignore wait list for now
+  (void)numEventsInWaitList;
+  (void)phEventWaitList;
+  //
+
+  ol_event_handle_t EventOut = nullptr;
+
+  void *DevPtr = std::get<BufferMem>(hBuffer->Mem).Ptr;
+
+  olMemcpy(hQueue->OffloadQueue, pDst, hQueue->Context->OffloadHost,
```

Comment:
Should we not be checking the result of this against `OL_SUCCESS`?

Reply:
Cool, I can do that in a separate PR